File: Release/WetiMatlabFunctions/FileOperations/ManipulateFastInputFile.m
% Function: ManipulateFastInputFile String replacement in FAST Input file
% -----------------------------
% Usage:
% n = ManipulateFastInputFile(TXTFile,Identifier,NewString)
% -----------------------------
% Input:
% TXTFile           String with name of .txt file
% Identifier        Identifier in FAST style input file 
% NewString         String with replacement
% -----------------------------
% Output:
% n                 number of replacements
% -----------------------------
% Created: 
% David Schlipf on 20-Dec-2022
% (c) WETI
% ----------------------------------
function n = ManipulateFastInputFile(TXTFile,Identifier,NewString)

[FOLDER,NAME,EXT]   = fileparts(TXTFile); 
TempTXTFile         = fullfile(FOLDER,[NAME,'_temp',EXT]);
fid                 = fopen(TXTFile);
fidTemp             = fopen(TempTXTFile,'w+');
n                   = 0;

while ~feof(fid)
    CurrentLine   	= fgetl(fid);
    % make sure only whole word are found (case sensitive)
    StartIdx        = regexp(CurrentLine, ['\s',Identifier,'(\s|$)'], 'start', 'once')+1;
    if isempty(StartIdx)
        fprintf(fidTemp,'%s\r\n',CurrentLine);
    else
        NewLine = [NewString,' ',CurrentLine(StartIdx:end)];
        fprintf(fidTemp,'%s\r\n',NewLine);
        n = n+1;
    end
end

fclose(fid);
fclose(fidTemp);
delete(TXTFile);
movefile(TempTXTFile,TXTFile);

File: Release/PythonFunctions/FileOperations/ManipulateFastInputFile.py
import os
import re

def ManipulateFastInputFile(TXTFile, Identifier, NewString):
    """
    String replacement in FAST Input file
    """
    FOLDER, NAME = os.path.split(TXTFile)
    NAME, EXT = os.path.splitext(NAME)
    TempTXTFile = os.path.join(FOLDER, NAME + '_temp' + EXT)

    n = 0
    with open(TXTFile, 'r') as fid, open(TempTXTFile, 'w+') as fidTemp:
        for line in fid:
            # make sure only whole word are found (case sensitive)
            match = re.search(r'\s' + Identifier + r'(\s|$)', line)
            if match is None:
                fidTemp.write(line)  # write the line as is, without adding a newline
            else:
                StartIdx = match.start() + 1
                NewLine = NewString + ' ' + line[StartIdx:]
                fidTemp.write(NewLine)  # write the new line, without adding a newline
                n += 1

    os.remove(TXTFile)
    os.rename(TempTXTFile, TXTFile)

    return n

